app/javascript/components/conditions/RejectionModal/component.js

```diff
@@ -0,0 +1,55 @@
+import React from 'react';
+import Modal from 'react-bootstrap/Modal';
+import Button from 'react-bootstrap/Button';
+import PropTypes from 'prop-types';
+import { inject, observer } from 'mobx-react';
+
+@inject('conditionSetStore')
+@observer
+class RejectionModal extends React.Component {
+  static propTypes = {
+    conditionSetStore: PropTypes.object,
+    title: PropTypes.string,
+    show: PropTypes.bool,
+    handleClose: PropTypes.func,
+  };
+
+  static defaultProps = {
+    title: 'Add Rejection Message',
+    show: false,
+  };
+
+  handleChange = (e) => {
+    const { conditionSetStore: { setRejectionMessage } } = this.props;
+    setRejectionMessage(e.target.value);
+  }
+
+  render() {
+    const { show, title, handleClose, conditionSetStore: { rejectionMsgTranslations } } = this.props;
+    return (
+      <Modal show={show} onHide={handleClose}>
+        <Modal.Header closeButton>
+          <Modal.Title>{title}</Modal.Title>
+        </Modal.Header>
+        <Modal.Body>
+          <div className="elmo-form">
+            <div className="form-field">
+              <label className="main" htmlFor="rejection-message">Message:</label>
+              <div className="control">
+                <div className="widget">
+                  <input id="rejection-message" className="form-control" type="text" value={rejectionMsgTranslations.en} onChange={this.handleChange} />
```

Comment:
As long as the DOM elements for this lie inside the Questioning form, this input will get included in the submission to the server. Which is why we don't need the hidden inputs elsewhere if we have the right name attrib on this.

This should become a loop over `rejectionMsgTranslations`, with one text field per translation. The name attrib should look like `${namePrefix}[rejection_msg_translations][${locale}]`.

Then on the server it will arrive as `{..., rejection_msg_translations: {en: 'blah', fr: 'foo'}, ...}`, which will be permitted with the change I suggested above.